sql/updates/world/4.3.4/2021_01_21_01_world.sql

```diff
@@ -99,10 +99,10 @@ UPDATE `creature_trainer` SET `TrainerId` = 127 WHERE `CreatureId` = 16502;
 
 -- Correct NPC Flags
 -- low level warrior
-UPDATE `creature_template` SET `npcFlag` = 51 WHERE `entry` = 38037;
-UPDATE `creature_template` SET `npcFlag` = 50 WHERE `entry` = 42324;
+UPDATE `creature_template` SET `npcFlag` = `npcFlag` | 0x01 | 0x10 | 0x20 WHERE `entry` = 38037;
+UPDATE `creature_template` SET `npcFlag` = `npcFlag` &~ 0x01 | 0x10 | 0x20 WHERE `entry` = 42324;
 -- low level shaman
-UPDATE `creature_template` SET `npcFlag` = 50, `trainer_class` = 7 WHERE `entry` = 38242;
+UPDATE `creature_template` SET `npcFlag` = `npcFlag` &~ 0x01 | 0x10 | 0x20 WHERE `entry` = 38242;
```

Comment:
are you sure you want to remove these flags instead of adding them? &~ substracts them. Just go with plain | to apply the flags.

Reply:
Yes, per my original comments in this PR:

> Trainers with gossip_menu_id 0 don't work correctly when npcFlag 1 is included.

Reply:
"Not work correctly" that is how?

Reply:
The gossips wouldn't show.

When I read that I thought of something I could try.  Apparently the OptionIndex needs to be set when npcFlag 1 is there.  Otherwise you can keep it at 0, like I did in this script.  I'll update the script to address that and remove the `&~ 0x01` code.